### ISSUE TITLE
Backspace not working with non-built in keyboard

### ISSUE THREAD

**Issue by dov** [tccowper/fabularium#3]
I recently got myself a an Android based E-book (Onyx Nova Pro) reader and I thought it would be nice to do some interactive fiction on it. I found fabularium and configured it to not use the internal keyboard, as I'm an old fan of the Grafitti keyboard, and I prefer using it to the internal keyboard.

Unfortunately I found that doing Backspace doesn't work for that keyboard, even though it works everywhere else. Testing various different keyboard showed that backspace works for keyboard based keyboards, but whenever there is handwriting involved it doesn't accept it. E.g. it doesn't recognize the backspace key for the Google Handwriting keyboard.

Btw, instead of replacing the whole keyboard, couldn't you do something more similar to the terminal emulator Termux that allows configuring an additional row of keys above whatever keyboard the user chooses? That seems less intrusive to me.

**Comment by KevinBlue18** [comment]
I agree as well, I emailed you this before kcblue14 at gmail do > t com. I'll try testing this on my fork using _Hacker's Keyboard_.

**Comment by cvshepherd** [comment]
it also doesn't work using the stock samsung keyboard on android 8.

**Comment by Sandman1973** [comment]
Hello. I have a related issue with the backspace. It's not quite the same, but...
I have a Gemini PDA with a physical keyboard which I thought would be idea for IF. All works fine but when the backspaces is pressed it deletes two characters! I wonder if the code is overriding something and passing it on when it shouldn't or preventing something when it should.

Perhaps the content of the function "public boolean dispatchKeyEvent(@NonNull KeyEvent event) {" in GLKActivity.java is the cause?

I installed Android Studio earlier, and downloaded the code, however it did not "just build". Tweaking a few things I managed to removed some build issues, but was unable to get it to build. Since I have zero experience with Android Studio it may be something in the environment that I need that is not there.

Has anyone managed to build this project? Has anyone had any contact with Tim? I was in contact with suggestions and feedback, but I've not heard anything from him for over two years now! Hope he is ok.

**Comment by npostavs** [comment]
Stock Samsung keyboard Android 8 is working fine for me.

I have some issues with 8Vim (a somewhat experimental soft-keyboard), mentioned here: flide/8VIM#173. It seems there are multiple ways for a keyboard to send input (as individual up & down key events, input strings), and backspace might be more complicated than that.

> For comparison:
> 
>   -  Hacker's Keyboard (https://github.com/klausw/hackerskeyboard) does send everything correctly.
>   -  Simple Keyboard (https://github.com/rkkr/simple-keyboard) sends all letters, numbers, etc, but the backspace key doesn't move the cursor (although the deletion does happen, which is revealed after sending new input).
>

**Comment by npostavs** [comment]
> Stock Samsung keyboard Android 8 is working fine for me. [...] Hacker's Keyboard does send everything correctly.

Ah, I just found out both of these only work correctly when they have completion enabled. I guess they maintain a text buffer internally for the completion and resend the text on each keypress?

**Comment by locrianz** [comment]
Same issue on the Supernote Manta with the handwriting keyboard.
